aiida/manage/configuration/options.py

```diff
@@ -8,252 +8,131 @@
-        'default': True,
-        'description': 'Boolean whether to print AiiDA deprecation warnings',
-        'global_only': False,
-    },
-    'transport.task_retry_initial_interval': {
-        'key': 'task_retry_initial_interval',
-        'valid_type': 'int',
-        'valid_values': None,
-        'default': 20,
-        'description': 'Initial time interval for the exponential backoff mechanism.',
-        'global_only': False,
-    },
-    'transport.task_maximum_attempts': {
-        'key': 'task_maximum_attempts',
-        'valid_type': 'int',
-        'valid_values': None,
-        'default': 5,
-        'description': 'Maximum number of transport task attempts before a Process is Paused.',
-        'global_only': False,
-    },
-}
-
-
-def get_option(option_name):
-    """Return a configuration option.configuration
-
-    :param option_name: the name of the configuration option
-    :return: the configuration option
-    :raises ValueError: if the configuration option does not exist
-    """
-    try:
-        option = Option(option_name, **CONFIG_OPTIONS[option_name])
-    except KeyError:
-        raise ValueError(f'the option {option_name} does not exist')
-    else:
-        return option
+import jsonschema
 
+from aiida.common.exceptions import ConfigurationError
 
-def get_option_names():
-    """Return a list of available option names.
+__all__ = ('get_option', 'get_option_names', 'parse_option', 'Option')
 
-    :return: list of available option names
-    """
-    return CONFIG_OPTIONS.keys()
+NO_DEFAULT = ()
 
 
-def parse_option(option_name, option_value):
+class Option:
+    """Represent a configuration option schema."""
+
+    def __init__(self, name: str, schema: Dict[str, Any]):
+        self._name = name
+        self._schema = schema
+
+    def __str__(self) -> str:
+        return f'Option(name={self._name})'
+
+    @property
+    def name(self) -> str:
+        return self._name
+
+    @property
+    def schema(self) -> Dict[str, Any]:
+        return self._schema
+
+    @property
+    def valid_type(self) -> Any:
+        return self._schema.get('type', None)
+
+    @property
+    def default(self) -> Any:
+        return self._schema.get('default', NO_DEFAULT)
+
+    @property
+    def description(self) -> str:
+        return self._schema.get('description', '')
+
+    @property
+    def global_only(self) -> bool:
+        return self._schema.get('global_only', False)
+
+    def validate(self, value: Any, cast: bool = True) -> Any:
+        """Validate a value
+
+        :param value: The input value
+        :param cast: Attempt to cast the value to the required type
+
+        :return: The output value
+        :raise: ConfigValidationError
+
+        """
+        # pylint: disable=too-many-branches
+        from .config import ConfigValidationError
+        from aiida.manage.caching import _validate_identifier_pattern
+
+        if cast:
+            try:
+                if self.valid_type == 'boolean':
```

Comment:
do we really have to do this ourselves or is there some existing library function we can use?

just as an example, I believe this will accept an empty list as a boolean (false) value.

Nothing against the logic implemented here (it seems to work for what I wanted to do) but I'm wondering whether there would be a way for us not to introduce yet another dialect of literals

Reply:
I copied this logic from the existing code, but am happy to consider an alternative

Reply:
ok, then let's keep it for the moment